src/v/redpanda/admin/api-doc/transform.json

```diff
@@ -45,6 +45,37 @@
         }
       ]
     },
+    {
+      "path": "/v1/transform/{name}/pause",
```

Comment:
BTW if you do go for the PATCH metadata endpoint instead I think you'll want to make a struct that is like:

```
struct transform_metadata_patch {
  absl::flat_hash_map<sstring, sstring> env; // clarify if this field has PATCH or PUT semantics I would vote for PUT
  bool is_paused;
  // Is there anything else here you can update? I don't think so.
};
```